probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+
+            numeric_transformer = Pipeline(
+                steps=[("imputer", self.strategies[strategy])]
+            )
+
+            categorical_transformer = Pipeline(
+                steps=[
+                    (
+                        "imp_cat",
+                        SimpleImputer(
+                            strategy="constant",
+                            fill_value="missing",
+                            add_indicator=True,
+                        ),
+                    ),
+                    ("ohe_cat", OneHotEncoder(handle_unknown="ignore")),
+                ]
+            )
+
+            preprocessor = ColumnTransformer(
+                transformers=[
+                    ("num", numeric_transformer, numeric_columns),
+                    ("cat", categorical_transformer, categorical_columns),
+                ],
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy=strategy
+            )
+
+            results.append(temp_results)
+
+        # If model supports missing values by default, then calculate the scores
+        # on raw data without any imputation.
+        if self.model_na_support:
+
+            categorical_transformer = Pipeline(
+                steps=[
+                    ("ohe_cat", OneHotEncoder(handle_unknown="ignore")),
+                ]
+            )
+
+            preprocessor = ColumnTransformer(
+                transformers=[("cat", categorical_transformer, categorical_columns)],
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy="No Imputation"
+            )
+            results.append(temp_results)
+
+        self.report_df = pd.DataFrame(results)
+        self.report_df.sort_values(by=f"{self.scorer.metric_name} score", inplace=True)
+        self.fitted = True
+        return self
+
+    def _calculate_results(self, X, y, clf, strategy):
+        """
+        Method to calculate the results for a particular imputation strategy.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            clf (binary classifier,sklearn.Pipeline):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategy(string):
+                Name of the strategy used for imputation.
+
+        Returns:
+
+            temp_df(dict) :  Dictionary containing the results of the evaluation.
+        """
+        imputation_results = cross_val_score(
```

Comment:
Please set the random seed, according to selt.random_seed

Reply:
I could not find random_state setting in cross_val_score. Do you know how I can do that ?

Reply:
You are right,  this one doesnt need to be set as long as we set the np random seed, which you do.